src/services/TestResultsService.ts

```diff
@@ -128,38 +139,83 @@ export class TestResultsService {
   public updateTestResult(systemNumber: string, payload: ITestResult, msUserDetails: IMsUserDetails) {
     this.removeNonEditableAttributes(payload);
     let validationSchema = this.getValidationSchema(payload.vehicleType, payload.testStatus);
-    validationSchema = validationSchema!.keys({countryOfRegistration: Joi.string().valid(countryOfRegistration).required().allow("", null)});
-    validationSchema = validationSchema!.optionalKeys(["testStationType", "testerEmailAddress", "testEndTimestamp", "systemNumber", "vin"]);
+    const testTypesValidationErrors = this.validateTestTypes(payload);
+    if (testTypesValidationErrors.length) {
+      return Promise.reject(new HTTPError(400, {errors: testTypesValidationErrors}));
+    }
+    // temporarily remove testTypes to validate only vehicle details and append testTypes to the payload again after the validation
+    const {testTypes} = payload;
+    delete payload.testTypes;
+    validationSchema = validationSchema!.keys({
+      countryOfRegistration: Joi.string().valid(COUNTRY_OF_REGISTRATION).required().allow("", null),
+      testTypes: Joi.any().forbidden()
+    });
+    validationSchema = validationSchema.optionalKeys(["testStationType", "testerEmailAddress", "testEndTimestamp", "systemNumber", "vin"]);
```

Comment:
Can be moved to a validationSchema file `updateTestResultSchema`

Reply:
This "validationSchema" is dynamically returned based on the vehicleType and testStatus and I'm extending and customising it. I don't think I understand why it should be moved into a separate file. At best we can have another function that alters the validationSchema to keep the update function cleaner. Thoughts?

Reply:
my mistake, overlooked the function on line 131